### PR TITLE
macos m1 support

### PR DESCRIPTION
**Pull request by benty-amzn**
This branch adds support for macos on aarch64 (m1)

See also #165

### CODE REVIEW

File: make/autoconf/build-aux/config.guess [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #!/bin/sh
 #
-# Copyright (c) 2012, Oracle and/or its affiliates. All rights reserved.
+# Copyright (c) 2012, 2021, Oracle and/or its affiliates. All rights reserved.
+# Copyright (c) 2021, Azul Systems, Inc. All rights reserved.
```
Comment (lutkerd) [inline make/autoconf/build-aux/config.guess:4]: Will this be removed when upstreamed?

File: make/lib/Awt2dLibraries.gmk [hunk 2]
```diff
@@ -433,7 +433,7 @@ $(eval $(call SetupJdkLibrary, BUILD_LIBJAVAJPEG, \
     OPTIMIZATION := HIGHEST, \
     CFLAGS := $(CFLAGS_JDKLIB), \
     HEADERS_FROM_SRC := $(LIBJPEG_HEADERS_FROM_SRC), \
-    DISABLED_WARNINGS_gcc := clobbered implicit-fallthrough shift-negative-value, \
+    DISABLED_WARNINGS_gcc := clobbered implicit-fallthrough shift-negative-value array_bounds, \
```
Comment (lutkerd) [inline make/lib/Awt2dLibraries.gmk]: This should be array-bounds. See https://bugs.openjdk.java.net/browse/JDK-8280999

File: installers/mac/pkg/build.gradle [hunk 3]
```diff
@@ -48,9 +48,7 @@ task inflatePkgTemplate {
         copy {
             from('templates/CorrettoPkg.pkgproj.template') {
                 rename { file -> file.replace('.template', '') }
-                project.version.each {
-                    prop -> filter(org.apache.tools.ant.filters.ReplaceTokens, tokens: [ (prop.key): String.valueOf(prop.value)])
-                }
+                filter(org.apache.tools.ant.filters.ReplaceTokens, tokens: ["full": project.version["full"], "major": project.version["major"], "arch": correttoArch, "macos-arch": (correttoArch == "aarch64" ? "arm64" : "x86_64")])
```
Comment (lutkerd) [inline installers/mac/pkg/build.gradle:51]: Can macos-arch just be nativeArch from toplevel build.gradle?

File: make/autoconf/build-aux/config.guess (hunk 1 above)
Comment (benty-amzn) [reply make/autoconf/build-aux/config.guess:4]: I don't expect it to be, no - if you grep through `develop` source for Azul Systems, Inc. there are a few instances dating back to 2016

File: installers/mac/pkg/build.gradle (hunk 3 above)
Comment (benty-amzn) [reply installers/mac/pkg/build.gradle:51]: no - we need `nativeArch` to follow existing convention and be `aarch64` for other places in our build logic. But for this substitution specifically it should be `arm64` instead.

File: make/lib/Awt2dLibraries.gmk (hunk 2 above)
Comment (benty-amzn) [reply make/lib/Awt2dLibraries.gmk]: Ack. It looks like this fix is already present in 11u-dev https://github.com/openjdk/jdk11u-dev/commit/85c81d9515ba2b98eda9442b3b07c29a230f2a56 so we should pick it up organically. I can cherry-pick that commit into this PR if you like.

**Review by phohensee — APPROVED** [review]
Lgtm.

**Comment by benty-amzn** [comment]
Rebased locally and pushed to develop

**Pull request closed without merging**